src/main/scala/com/cognite/sdk/scala/v1/resources/dataPoints.scala

```diff
@@ -291,34 +242,47 @@ class DataPointsResourceV1[F[_]](val requestSession: RequestSession[F])
       granularity: String,
       aggregates: Seq[String]
   ): F[Map[String, Seq[DataPoint]]] =
-    requestSession
-      .sendCdf(
-        { request =>
-          request
-            .post(uri"$baseUri/list")
-            .body(
-              Items(
-                Seq(
-                  QueryAggregatesById(
-                    id,
-                    inclusiveStart.toString,
-                    exclusiveEnd.toString,
-                    granularity,
-                    aggregates
-                  )
-                )
-              )
-            )
-            .response(asProtobufOrCdpApiError)
-            .mapResponse {
-              case Left(value) => throw value.error
-              case Right(Left(cdpApiError)) => throw cdpApiError.asException(uri"$baseUri/list")
-              case Right(Right(dataPointListResponse)) =>
-                toAggregateMap(parseAggregateDataPoints(dataPointListResponse))
-            }
-        },
-        accept = "application/protobuf"
+    queryProtobuf(
+      Items(
+        Seq(
+          QueryRangeById(
+            id,
+            inclusiveStart.toString,
+            exclusiveEnd.toString,
+            None,
+            Some(granularity),
+            Some(aggregates)
+          )
+        )
       )
+    ) { dataPointListResponse =>
+      toAggregateMap(parseAggregateDataPoints(dataPointListResponse))
+    }
+
+  def queryAggregatesByIdWithLimit(
```

Comment:
As discussed it could be nicer for the user to have limit as an optional parameter in the `queryAggregatesById` and similar functions.

Reply:
Roger that but I'd like to do that as a separate PR if we decide to go for that approach.